src/templates/index.jsx

```diff
@@ -125,6 +126,12 @@ const Layout = React.createClass({
     },
 
     componentWillReceiveProps: function(nextProps) {
+        if (this.props.selectedNoteId) {
+            this.setState({
+                previouslySelectedNote: this.props.selectedNoteId,
```

Comment:
can we keep the `id` terminology the same so that we don't confuse objects with ids?

we have made this mistake in this app before and it's unfortunately caused much grief. that is to say that we have had situations where we had a valid note object but it wasn't in the list so the id was useless. I would much prefer here to track just the id and indicate via the name that such is the case: `previouslySelectedNoteId` or something